.github/workflows/tarpaulin.yml

```diff
@@ -0,0 +1,66 @@
+name: tarpaulin
+
+on:
+  pull_request:
+    paths:
+      - Cargo.lock
+      - Cargo.toml
+      - src/*.rs
+      - src/**/*.rs
+      - src/**/**/*.rs
+
+permissions:
+  contents: read
+  pull-request: write
+
+jobs:
+  coverage:
+    name: coverage
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: python
+        uses: actions/setup-python@v4.5.0
+        with:
+          python-version: 3.11
+
+      - name: checkout
+        uses: actions/checkout@v3.5.0
+        with:
+          persist-credentials: false
+          ref: ${{ github.head_ref }}
+
+      - name: cache
```

Comment:
Tarpaulin takes a while to install.  Caching it is thus a good idea.  We name the cache based on the commit number which triggered this workflow for future reference.  This causes the cache to never be exactly matched, so we need to be careful when relying on the restored cache.

Reply:
If the cache key is based on the commit sha? Which example scenario would have a cache hit?

Reply:
Well, it was originally designed to make sure that newer versions of Tarpaulin are built while the older ones are ignored.  This did not work as expected, so I will change it.

With the `restore-keys` clause of the Action, partial cache hits are achieved as the initial name parts are the same.  The Action will then choose the latest available cache.  Since the hit was only partial, however, an exact hit is impossible.  But at the moment, restoring caches does not work at all due to a critical bug in the Action definition, see actions/cache#1152.

Reply:
With your go I'll merge it 🙂👍

Reply:
Okay, you can merge it.